docs/docs/query-filters.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Query Filters with GraphQL in Gatsby
+---
+
+> This documentation is up to date as of `gatsby@2.23.0`.
+
+## Summary
+
+Gatsby stores all data loaded during the source-nodes phase in Redux and it allows you to write GraphQL queries to query that data. This data, stored as individual "nodes", can be searched through using a query language that is a subset of [MongoDb queries](https://docs.mongodb.com/manual/reference/operator/query/).
+
+When you write a GraphQL query, the `filter` parts will be processed and Gatsby will return all the nodes that match each of the given filters. GraphQL does the rest.
```

Comment:
```suggestion
Filtering is used in GraphQL root fields of Node types (e.g. for File type it would be `file` and `allFile`). `filter` GraphQL argument is passed to the filtering system and will return all the nodes that match each of the given filters. The resto of the processing, such as pagination, is handled on GraphQL resolver level.
```